mountpoint-s3/src/prefetch.rs

```diff
@@ -273,6 +275,10 @@ where
                     "out-of-order read, resetting prefetch"
                 );
                 counter!("prefetch.out_of_order", 1);
+                histogram!(
+                    "prefetch.contiguous_read_len",
```

Comment:
Alternative option: what if we logged the "delta" offset? i.e. how far away the seek is?
It could be useful data to compare to the seek windows, for example.

Reply:
> Could we implement Drop for PrefetchGetObject and log it there?

Exactly what I was planning, but wanted to discuss if we think this is a useful metric first.

Reply:
discussed: probably it would be good to have both since they provide different information.

Not urgent to follow up with this right now

Reply:
I like this metric, but agree that we need to emit it at drop time too.

Reply:
I'm going to push an update to get this ready to merge. Implementing drop does feel simplest.

Reply:
>Alternative option: what if we logged the "delta" offset? i.e. how far away the seek is?
>It could be useful data to compare to the seek windows, for example.

I see we almost already have this with the `prefetch.seek_distance` histogram (forward/backward recorded with `dir` dimension). The bit that's missing is that it isn't recorded if the seek is not within the seek window - but maybe we just start recording it regardless. I'll leave that for a separate PR though.

https://github.com/awslabs/mountpoint-s3/blob/83fa7c654623cba34b44f99c3b8930b693ebd0b1/mountpoint-s3/src/prefetch.rs#L498

https://github.com/awslabs/mountpoint-s3/blob/83fa7c654623cba34b44f99c3b8930b693ebd0b1/mountpoint-s3/src/prefetch.rs#L520

Reply:
>"contiguous_read_len" could be misleading because it does not account for try_seek. Or do we think it's still a good enough approximation?

Added a comment to capture the reasoning here.